src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -175,6 +185,44 @@ CitusExecScan(CustomScanState *node)
  */
 static void
 CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
+{
+	CitusScanState *scanState = (CitusScanState *) node;
+	DistributedPlan *distributedPlan = scanState->distributedPlan;
+
+	/*
+	 * If we have not already copied the plan into this context, do it now.
+	 * Note that we could have copied the plan during CitusGenerateDeferredQueryStrings.
+	 */
+	if (GetMemoryChunkContext(distributedPlan) != CurrentMemoryContext)
+	{
+		distributedPlan = copyObject(distributedPlan);
+
+		scanState->distributedPlan = distributedPlan;
```

Comment:
I'm wondering whether we should only copy the Job and put it in the CitusScanState (e.g. jobToExecute). That way, the job in the DistributedPlan becomes more like a template of the Job at execution time.

Reply:
Do you prefer for me to work on that in a separate PR on top of #3388?

Reply:
Maybe on top of #3388.